HeartbeatLogging/Sources/_ObjC_HeartbeatController.swift

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// An object that provides API to log and flush heartbeats from a synchronized storage container.
+@objc(FIRHeartbeatController)
+@objcMembers
+public class _ObjC_HeartbeatController: NSObject {
+  /// The underlying Swift object.
+  private let heartbeatController: HeartbeatController
```

Comment:
The heartbeat logging module has two native swift types that need to be bridged over for use outside module (i.e. FirebaseCore).

I prefixed the swift type with _ObjC_ to "hide" it from Swift clients.

Reply:
Why not use `NS_REFINED_FOR_SWIFT`? https://developer.apple.com/documentation/swift/objective-c_and_c_code_customization/improving_objective-c_api_declarations_for_swift

Reply:
Because `NS_REFINED_FOR_SWIFT` is for ObjC → Swift. 

For Swift → ObjC, types need to be public and conform to `NSObject`.

Reply:
Some extra context.

I tried making `HeartbeatController` class conform to `NSObject` to make it objc-compatible. But, the `flush` API returns a `HeartbeatsPayload` which is a Swift struct and does not bridge. So I had to create a bridging type for `HeartbeatsPayload`.

We could eliminate the need for these custom bridging types by turning `HeartbeatController` and `HeartbeatsPayload` into `NSObject`s but I felt `HeartbeatsPayload` should stay a value type. OTOH, `HeartbeatsPayload` is immutable as it is so making it a reference type wouldn't affect its usage.

WDYT?

Reply:
But wouldn't `NS_REFINED_FOR_SWIFT` also hide the Objective C API from Swift so it wouldn't need to be prefixed with the underscore?

Reply:
Where are you suggesting using `NS_REFINED_FOR_SWIFT`?

Reply:
Thanks. I see my confusion now. It make sense.